modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskService.java

```diff
@@ -17,7 +17,17 @@
 public interface TaskService {
 
     /**
-     * Saves the given task in the database.
+     * Saves the given task in the database giving choice whether
+     * register its handler or not.
+     *
+     * @param task  the task to be saved, not null
+     * @param registerHandler   if true the tasks handler will be registered
+     *                          if false it will not
+     */
+    void save(final Task task, boolean registerHandler);
```

Comment:
I don't see much point in this method. Why would I want to register a task without registering a handler? Doesn't make sense. If this is in order to prevent double scheduling of the listener for scheduled tasks, that can be done on the implementation level.